models/comments.model.js

```diff
@@ -0,0 +1,22 @@
+const db = require('../db/connection');
+
+exports.fetchCommentsByArticleId = (article_id) => {
+  return db
+    .query(
+      `SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC`,
+      [article_id]
+    )
+    .then((result) => {
+      if (result.rows.length === 0) {
+        return db
+          .query('SELECT * FROM articles WHERE article_id = $1', [article_id])
+          .then((articleCheck) => {
+            if (articleCheck.rows.length === 0) {
+              return Promise.reject({ status: 404, msg: 'Article not found' });
+            }
+            return [];
+          });
+      }
+      return result.rows;
+    });
```

Comment:
This is a good start, but in its current form is difficult to read and tell whats happening at a glance. You have a nested promise, to check if the article exists, but this can be done at the same time as querying for the comments by a the article_id. I would recommend using a Promise.all(...) here instead.